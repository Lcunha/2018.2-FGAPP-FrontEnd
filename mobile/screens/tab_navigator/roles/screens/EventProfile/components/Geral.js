import React, { Component } from "react";
import {
	Card,
	CardItem,
	Left,
	Thumbnail,
	Body,
	H2,
} from "native-base";
import { Text, StyleSheet } from "react-native";
import Divider from "./Divider";
import GeralDetails from "./GeralDetails";
import * as helpers from '../../../utils/helpers';

const noPic = require("../../../static/noPic.png");
const adultOnly = require("../../../static/adultOnly.png");

class Geral extends Component {
	render() {
		const uri = this.props.photo;
		return (
			<Card>
				<CardItem>
					<Left>
						<Thumbnail
							style={styles.thumbnailGeral}
							source={this.props.photo == null ? noPic : { uri: uri }}
						/>
					</Left>

					<Body>
						<H2
							style={{
								textAlign: "center",
								alignSelf: "center"
							}}
						>
							{this.props.eventName}
						</H2>

						<Divider />
            
            <GeralDetails iconName="time" text={`${this.props.eventHour.slice(0,5)}h`}/>

            <GeralDetails iconName="calendar" text={ helpers.formatDate(this.props.eventDate).formatted }/>

            <GeralDetails iconName="cash" text={ `R$ ${this.props.value}` }/>

					</Body>
				</CardItem>

				{/* Sessão do +18; Se for true, ele mostra o card dizendo que a classificação indicativa é +18 */}
				{this.props.adultOnly == true && (
					<Card transparent>
						<Body>
							<CardItem>
								<Text>Classificação Indicativa:</Text>
								<Thumbnail
									style={{ marginLeft: 20 }}
									small
									source={adultOnly}
								/>
							</CardItem>
						</Body>
					</Card>
				)}
			</Card>
		);
	}
}

const styles = StyleSheet.create({
	descCards: {
		flexDirection: "row"
	},
	thumbnailGeral: {
		borderWidth: 1,
		borderColor: "grey",
		borderRadius: 125,
		width: 140,
		height: 140
	}
});

export default Geral;
