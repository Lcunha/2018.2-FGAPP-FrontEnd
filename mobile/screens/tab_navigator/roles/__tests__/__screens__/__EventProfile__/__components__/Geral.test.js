import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Geral from '../../../../screens/EventProfile/components/Geral';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<Geral />).toJSON();
  expect(tree).toMatchSnapshot();
});