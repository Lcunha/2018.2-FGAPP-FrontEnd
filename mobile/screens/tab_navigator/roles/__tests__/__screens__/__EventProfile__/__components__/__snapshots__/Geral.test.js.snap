// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
  virtual={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "grey",
        "marginBottom": 10,
      }
    }
  >
    Geral
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
      virtual={undefined}
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            },
            Object {
              "borderColor": "grey",
              "borderRadius": 125,
              "borderWidth": 1,
              "height": 140,
              "width": 140,
            },
          ]
        }
        virtual={undefined}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 24,
              "lineHeight": 27,
            },
            Object {
              "alignSelf": "center",
              "textAlign": "center",
            },
          ]
        }
        virtual={undefined}
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "borderWidth": 0.25,
            "margin": 20,
            "opacity": 0.25,
            "width": 100,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "flexDirection": "row",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Hora inv√°lida
            h
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "flexDirection": "row",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "flexDirection": "row",
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            R$ 
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
